backend/server/src/main/java/ai/verta/modeldb/authservice/MDBRoleServiceUtils.java

```diff
@@ -318,14 +319,10 @@ public String buildReadOnlyRoleBindingName(
   @Override
   public Workspace getWorkspaceByWorkspaceName(
       UserInfo currentLoginUserInfo, String workspaceName) {
-    /*from the name for workspace, get the workspace id and type.
-    if no workspace is present assume user's personal workspace*/
-    if (workspaceName == null || workspaceName.isEmpty()) {
-      return authService.workspaceIdByName(
-          true, authService.getUsernameFromUserInfo(currentLoginUserInfo));
-    } else {
-      return authService.workspaceIdByName(true, workspaceName);
+    if (Strings.isNullOrEmpty(workspaceName)) {
+      return Workspace.newBuilder().build();
```

Comment:
I'm not sure I follow this.  From the description in the PR:

> The decision on which workspace to use if workspace not specified should be handled by UAC

but it looks like here you're returning an empty workspace, not letting UAC handle it. Can you clarify?

Reply:
This method is to set request parameters during setResource or getResource call. We should not set in MDB some workspace in request and we can set default workspace by calling Workspace.newBuilder().build().